osu.Game/Screens/Play/Player.cs

```diff
@@ -383,7 +383,7 @@ public override bool OnExiting(IScreen next)
                 return true;
             }
 
-            if ((!AllowPause || HasFailed || !ValidForResume || pauseContainer?.IsPaused != false || RulesetContainer?.HasReplayLoaded != false) && (!pauseContainer?.IsResuming ?? true))
+            if ((!AllowPause || HasFailed || !ValidForResume || pauseContainer?.IsPaused.Value != false || RulesetContainer?.HasReplayLoaded?.Value != false) && (!pauseContainer?.IsResuming ?? true))
```

Comment:
What is this change? How can `HasReplayLoaded` ever be null? How does this "Fix crashes on beatmap not being set correctly in player"?

Reply:
Check the initialisation code for Player. It exits early.

Reply:
This has nothing to do with initialisation. This is checking whether some bindable member of container is null, which it can never be?

Reply:
Definitely can be, since it happened in a test case run.

Reply:
I'm 10000% sure it can't be:


<img width="506" alt="image" src="https://user-images.githubusercontent.com/1329837/52682890-5af98000-2f84-11e9-9ba7-f50e418ea3aa.png">

Reply:
as discussed, other one